packages/frontend/src/utils/getUser.ts

```diff
@@ -1,14 +1,43 @@
-import { doc, getDoc } from 'firebase/firestore';
+import {
+  collection,
+  doc,
+  getDoc,
+  getDocs,
+  query,
+  where,
+} from 'firebase/firestore';
 import { firebaseDb } from 'src/main';
 import { IUser } from 'src/types';
+import { editUser } from './editUser';
 
-export const getUser = async (telegramName: string) => {
+export const getUser = async (telegramName: string, telegramId?: number) => {
   const docRef = doc(firebaseDb, 'users', telegramName);
   const docSnap = await getDoc(docRef);
 
   if (docSnap.exists()) {
     return docSnap.data() as IUser;
-  } else {
-    return undefined
   }
+
+  if (telegramId) {
+    const queryForUserById = query(
+      collection(firebaseDb, 'users'),
+      where('telegramId', '==', telegramId)
+    );
+
+    const querySnapshot = await getDocs(queryForUserById);
+
+    if (querySnapshot.size === 1) {
+      querySnapshot.forEach((docSnap) => {
+        const user = docSnap.data() as IUser;
+
+        editUser(user.telegramName, { telegramName }, false);
```

Comment:
in this case we will only change field telegramName, but document id will still be his initial user name, won't it? And this will happen every login for users that have ever changed a name. Maybe it's better to recreate user in such case?

Reply:
in editUser function has different case for changing telegramName:
<img width="556" alt="image" src="https://github.com/epam/epmgcip-uva.mono/assets/65450338/ca22ce28-1e8f-4927-8335-cfd1dc3bf33f">
In this case creates new user with updated key and telegramName field. Old user will delete